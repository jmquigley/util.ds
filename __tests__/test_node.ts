'use strict';

import {Color, Node} from '../index';

test('Test creation of a simple Node', () => {
	const node = new Node<string>('test');

	expect(node).toBeDefined()
	expect(node).toMatchSnapshot();
});

test('Test creation of a Node with right/left child', () => {
	const node = new Node<string>(
		'inner',
		new Node<string>('right child'),
		new Node<string>('left child'),
		new Node<string>('parent reference'),
		Color.black
	);

	expect(node).toBeDefined()
	expect(node).toMatchSnapshot();
});

test('Test clearing a Node', () => {
	const node = new Node<string>(
		'inner',
		new Node<string>('right child'),
		new Node<string>('left child'),
		new Node<string>('parent reference'),
		Color.black
	);

	expect(node).toBeDefined()
	expect(node.color).toBeTruthy();
	expect(node.data).toBeTruthy();
	expect(node.parent).toBeTruthy();
	expect(node.left).toBeTruthy();
	expect(node.right).toBeTruthy();

	node.clear();

	expect(node.color).toBeFalsy();
	expect(node.data).toBeFalsy();
	expect(node.parent).toBeFalsy();
	expect(node.left).toBeFalsy();
	expect(node.right).toBeFalsy();
});
