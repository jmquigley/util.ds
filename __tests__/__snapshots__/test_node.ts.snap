// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creation of an empty/null node 1`] = `
Node {
  "_options": Object {
    "children": Array [],
    "color": 1,
    "data": null,
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
  "children": Array [],
  "color": 1,
  "data": null,
  "id": null,
  "left": null,
  "parent": null,
  "parentId": null,
  "right": null,
}
`;

exports[`Set node elements after creation 1`] = `
Node {
  "_options": Object {
    "children": Array [],
    "color": 1,
    "data": "inner",
    "id": 12345,
    "left": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "left child",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "left child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "parent": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "parent reference",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "parent reference",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "parentId": null,
    "right": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "right child",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "right child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
  },
  "children": Array [],
  "color": 1,
  "data": "inner",
  "id": 12345,
  "left": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "left child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "left child",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
  "parent": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "parent reference",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "parent reference",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
  "parentId": null,
  "right": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "right child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "right child",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
}
`;

exports[`Set node elements after creation 2`] = `
Node {
  "_options": Object {
    "children": Array [],
    "color": 1,
    "data": "inner",
    "id": 12345,
    "left": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "left child",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "left child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "parent": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "parent reference",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "parent reference",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "parentId": null,
    "right": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "right child",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "right child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
  },
  "children": Array [],
  "color": 0,
  "data": "changed inner",
  "id": 98765,
  "left": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "left child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "left child",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
  "parent": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "parent reference",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "parent reference",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
  "parentId": 8675309,
  "right": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "right child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "right child",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
}
`;

exports[`Test creation of a Node with right/left child 1`] = `
Node {
  "_options": Object {
    "children": Array [],
    "color": 1,
    "data": "inner",
    "id": null,
    "left": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "left child",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "left child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "parent": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "parent reference",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "parent reference",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "parentId": null,
    "right": Node {
      "_options": Object {
        "children": Array [],
        "color": 1,
        "data": "right child",
        "id": null,
        "left": null,
        "parent": null,
        "parentId": null,
        "right": null,
      },
      "children": Array [],
      "color": 1,
      "data": "right child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
  },
  "children": Array [],
  "color": 1,
  "data": "inner",
  "id": null,
  "left": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "left child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "left child",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
  "parent": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "parent reference",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "parent reference",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
  "parentId": null,
  "right": Node {
    "_options": Object {
      "children": Array [],
      "color": 1,
      "data": "right child",
      "id": null,
      "left": null,
      "parent": null,
      "parentId": null,
      "right": null,
    },
    "children": Array [],
    "color": 1,
    "data": "right child",
    "id": null,
    "left": null,
    "parent": null,
    "parentId": null,
    "right": null,
  },
}
`;

exports[`Test creation of a simple Node 1`] = `
Node {
  "_options": Object {
    "children": Array [],
    "color": 1,
    "data": "test",
    "id": 12345,
    "left": null,
    "parent": undefined,
    "parentId": 67890,
    "right": null,
  },
  "children": Array [],
  "color": 1,
  "data": "test",
  "id": 12345,
  "left": null,
  "parent": undefined,
  "parentId": 67890,
  "right": null,
}
`;
